add support for doing a passthrough of requests to preferred backend for unregistered routes (#3018)
cortexproject/cortex: docs/operations/query-tee.md

- `/api/v1/alerts` (GET)
- `/api/v1/rules` (GET)

> #### PassThrough requests

Reviewer: ```suggestion
#### Pass-through requests
```

> `query-tee` supports acting as a PassThrough for requests to routes not matching any of the documented `API endpoints above`.

Reviewer: ```suggestion
`query-tee` supports acting as a transparent proxy for requests to routes not matching any of the documented API endpoints above.
```

When enabled, those requests are passed on to just the configured preferred backend.

> To activate this feature it requires setting `-proxy.passthrough-non-registered-routes` flag to `true` and configuring a preferred backend.

Reviewer: ```suggestion
To activate this feature it requires setting `-proxy.passthrough-non-registered-routes=true` flag and configuring a preferred backend.
```
